qa/test-harness/testharness.py

```diff
@@ -0,0 +1,159 @@
+
+        if msg_queue.qsize() == 0:
+            raise TestHarnessException("[ERROR] Aborting test routine! Received no messages from the Kafka consumer.")
+
+        # After all messages were received, log them to a file
+        self.validation_results = self.validator.validate_queue(msg_queue)
+        self.bundle_id = self.validation_results[0].record['metadata']['serialId']['bundleId']
+        self.serial_number = self.validation_results[0].record['metadata']['serialId']['serialNumber']
+
+        # Count the number of validations and failed validations
+        self.num_errors = 0
+        self.num_validations = 0
+        for result in self.validation_results:
+            self.num_validations += len(result.field_validations)
+            for validation in result.field_validations:
+                if validation.valid == False:
+                    self.num_errors += 1
+
+
+        if self.num_errors > 0:
+            print('[FAILED] ============================================================================')
+            print('[FAILED] Validation has failed! Detected %d errors out of %d total validation checks.' % (self.num_errors, self.num_validations))
+            print('[FAILED] ============================================================================')
+        else:
+            print('[SUCCESS] ===========================================================================')
+            print('[SUCCESS] Validation has passed. Detected no errors out of %d total validation checks.' % (self.num_validations))
+            print('[SUCCESS] ===========================================================================')
+
+        # Print the validation results to a file if the user has specified one
+        if self.output_file_path is not None:
+            self.print_results_to_file()
+            print("[END] Results logged to '%s'." % self.output_file_path)
+        else:
+            print("[END] Output file not specified, detailed results not logged.")
+
+        print("[END] File validation complete.")
+
+    def print_results_to_file(self):
+        logger = logging.getLogger('test-harness')
+        logger.setLevel(logging.INFO)
+        logger.addHandler(logging.FileHandler(self.output_file_path, 'w'))
+        printer = ValidationResultPrinter()
+        printer.print_list(self.validation_results, True, logger)
+
+
+class TestHarness:
+    def __init__(self, config_file_path):
+        config = configparser.SafeConfigParser(os.environ)
+        config.read(config_file_path)
+
+        self.test_harness_iterations = []
+        for key in config.sections():
+            if key == "_meta":
+                self.perform_bundle_id_check = True if config[key]['PerformBundleIdCheck'] == 'True' else False
+            else:
+                self.test_harness_iterations.append(TestHarnessIteration(config[key]))
+
+    def run(self):
+        bundle_streams = {}
+        for iteration in self.test_harness_iterations:
```

Comment:
`bunlde_streams` can probably be populated (appended) during the iteration run so we wouldn't need to specify in the config file.

Reply:
Done.